### PR TITLE
Feat: Product Page with Sorting and filters

### PR DESCRIPTION
**Pull request by Logan1x**
Major Changes
- Product listing
- sorting functionality working
- filters are working
- Added context, custom hooks, reducers
- folder structure improved

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -13,6 +14,7 @@ function App() {
       <Routes>
         <Route path="/" element={<Home />} />
         <Route path="products" element={<Products />} />
```
Comment (mdarif6) [inline src/App.jsx]: Route Path works with out "/", but I think it is a good practice to use forward slash.

**Review by mdarif6 — APPROVED** [review]
- rest folder structure is okay.
- you can give some name to alt=""

**Comment by parthpandyappp** [comment]
The folder structure is so organized in terms of composing the `Reducers`,  `Custom hooks` & `Contexts`. The setup really looks so clean. Go ahead with merge :+1:

**Review by parthpandyappp — APPROVED** [review]

File: src/pages/products.jsx [hunk 2]
```diff
@@ -1,70 +1,145 @@
...
           <h4>Sort By</h4>
           <form action="" className="sort-selection-radio">
             <div>
-              <input type="radio" id="priceHigh" name="sort" />
+              <input
+                type="radio"
+                id="priceHigh"
+                name="sort"
+                onChange={() =>
+                  dispatch({ type: "SORT_BY", payload: "HIGH_TO_LOW" })
```
Comment (Rohanmond) [inline src/pages/products.jsx:115]: instead of hardcoded string you can use constants

**Review by Rohanmond — APPROVED** [review]
Everything looks good

**Comment by Logan1x** [comment]
Thanks Everyone, merging now :)

**Pull request merged**